lib/src/native/realm_core.dart

```diff
@@ -2907,45 +2907,63 @@ extension on Pointer<realm_value_t> {
     if (this == nullptr) {
       throw RealmException("Can not convert nullptr realm_value to Dart value");
     }
+    return ref.toDartValueByRef(realm);
+  }
+}
 
-    switch (ref.type) {
+extension on realm_value_t {
+  Object? toDartValueByRef(Realm? realm) {
```

Comment:
The name could be just `toDartValue` 
`ByRef` only makes it confusing.

Reply:
Renamed